python_modules/libraries/dagster-dg/dagster_dg/templates/deploy_uv_editable_Dockerfile.jinja

```diff
@@ -0,0 +1,43 @@
+# Based on: https://github.com/astral-sh/uv-docker-example/blob/main/multistage.Dockerfile
+
+# First, build the application in the `/app` directory.
+FROM ghcr.io/astral-sh/uv:python{{ python_version }}-bookworm-slim AS builder
+ENV UV_COMPILE_BYTECODE=1 UV_LINK_MODE=copy
+
+# Disable Python downloads, because we want to use the system interpreter
+# across both images. If using a managed Python version, it needs to be
+# copied from the build image into the final image; see `standalone.Dockerfile`
+# for an example.
+ENV UV_PYTHON_DOWNLOADS=0
+
+WORKDIR /app
+
+COPY --from=oss python_modules ../dagster/python_modules
+COPY --from=internal dagster-cloud/python_modules/dagster-cloud ../dagster-internal/dagster-cloud/python_modules/dagster-cloud
+COPY --from=internal dagster-cloud/python_modules/dagster-cloud-cli ../dagster-internal/dagster-cloud/python_modules/dagster-cloud-cli
+
+RUN --mount=type=cache,target=/root/.cache/uv \
+    --mount=type=bind,source=uv.lock,target=uv.lock \
+    --mount=type=bind,source=pyproject.toml,target=pyproject.toml \
+    uv sync --frozen --no-install-project --no-dev
+ADD . /app
+RUN --mount=type=cache,target=/root/.cache/uv \
+    uv sync --frozen --no-dev
+
+
+# Then, use a final image without uv
+FROM python:{{ python_version }}-slim-bookworm
+# It is important to use the image that matches the builder, as the path to the
+# Python executable must be the same.
+
+# Copy the application from the builder
+COPY --from=builder --chown=app:app /app /app
+
+# And also the editable installs
+COPY --from=builder --chown=app:app /dagster /dagster
+COPY --from=builder --chown=app:app /dagster-internal /dagster-internal
```

Comment:
nit: where is the user app created?

To avoid running as root, I believe we're missing these two statements:
```
RUN groupadd -r app && useradd -r -g app app
```

and

```
USER app
```

the app user doesn't exist by default on the base image.

```shell
docker run -it --rm python:3.12-slim-bookworm bash -c "su app"
su: user app does not exist or the user entry does not contain all the required fields
```

Reply:
hm I pulled this directly from the astral example - maybe i should just take out those chown examples? My inclination is to keep the default templated dockerfile as simple as possible